### PR TITLE
CANTINA-958: Add better filtering in `wp vip two-factor report` CLI command

### PR DESCRIPTION
**Pull request by rebeccahum**
## Description

Adds filtering by whether 2fa is enabled, 2fa provider, user role, and user.

Example usages:
	 *      wp vip two-factor report
	 *      wp vip two-factor report --2fa-enabled=true
	 *      wp vip two-factor report --2fa-provider=email
	 *      wp vip two-factor report --role=administrator
	 *      wp vip two-factor report --role=administrator --2fa-enabled=false
	 *      wp vip two-factor report --role=administrator --2fa-enabled=true --2fa-provider=email
	 *      wp vip two-factor report --user_login=wpvip
	
### WP-CLI Command Updated: wp vip two-factor report

Add filtering by whether 2fa is enabled, 2fa provider, user role, and user.

## Pre-review checklist

Please make sure the items below have been covered before requesting a review:

- [x] This change works and has been tested locally (or has an appropriate fallback).
- [x] This change works and has been tested on a Go sandbox.
- [ ] This change has relevant unit tests (if applicable).
- [ ] This change uses a rollout method to ease with deployment (if applicable - especially for large scale actions that require writes).
- [ ] This change has relevant documentation additions / updates (if applicable).
- [x] I've created a changelog description that aligns with the provided examples.

## Pre-deploy checklist

- [ ] VIP staff: Ensure any alerts added/updated conform to internal standards (see internal documentation). 

## Steps to Test
1) Create users with different roles, administrative users with 2FA and administrative users without 2FA
2) Do the examples in description and ensure it returns the expected results

### CODE REVIEW

**Review by WPprodigy — APPROVED** [review]

**Comment by rebeccahum** [comment]
@GaryJones Can you plz traumatize me with a code review thx

File: wp-cli/vip-two-factor.php [hunk 1]
```diff
@@ -1,35 +1,130 @@
...
+	 *
+	 * ## EXAMPLES
+	 *
+	 *      wp vip two-factor report
+	 *      wp vip two-factor report --2fa-enabled=true
+	 *      wp vip two-factor report --2fa-provider=email
+	 *      wp vip two-factor report --role=administrator
+	 *      wp vip two-factor report --role=administrator --2fa-enabled=false
+	 *      wp vip two-factor report --role=administrator --2fa-enabled=true --2fa-provider=email
+	 *      wp vip two-factor report --user_login=wpvip
```
Comment (GaryJones) [inline wp-cli/vip-two-factor.php]:
```suggestion
	 *     # List 2FA status for all users.
	 *     $ wp vip two-factor report
	 *
	 *     # List users with 2FA enabled. 
	 *     $ wp vip two-factor report --2fa-enabled=true
	 *
	 *     # List users who use email as their authentication factor. 
	 *     $ wp vip two-factor report --2fa-provider=email
	 *
	 *     # List 2FA status for administrators. 
	 *     $ wp vip two-factor report --role=administrator
	 *
	 *     # List administrators who don't have 2FA enabled. 
	 *     $ wp vip two-factor report --role=administrator --2fa-enabled=false
	 *
	 *     # List 2FA status for administrators using the email authentication factor. 
	 *     $ wp vip two-factor report --role=administrator --2fa-enabled=true --2fa-provider=email
	 *
	 *     # List 2FA status only for the wpvip user. 
	 *     $ wp vip two-factor report --user_login=wpvip
```

Question:
> List users who use email as their authentication factor. 

Is this just primary factor, or does it include secondary providers as well?

File: wp-cli/vip-two-factor.php [hunk 2]
```diff
@@ -1,35 +1,130 @@
...
+	 *
+	 * @subcommand report
```
Comment (GaryJones) [inline wp-cli/vip-two-factor.php]:
```suggestion
```

Only needed when the command doesn't match the method name (i.e. hyphens vs underscore).

File: wp-cli/vip-two-factor.php [hunk 3]
```diff
@@ -1,35 +1,130 @@
...
 			}
+
+			if ( $twofa_enabled_flag && $user->two_factor_enabled !== $twofa_enabled_flag ) {
+				unset( $users[ $idx ] );
+			}
 		}
 
+		$fields   = array( 'ID', 'display_name', 'roles' );
 		$fields[] = 'two_factor_enabled';
 		$fields[] = 'two_factor_providers';
```
Comment (GaryJones) [inline wp-cli/vip-two-factor.php:141]:
```suggestion
		$fields   = array( 'ID', 'display_name', 'roles', 'two_factor_enabled', 'two_factor_providers' );
```

Further, looking at the output from ` wp user list --format=table`:

```sh
+----+------------------+-------------------+-----------------------------------------+---------------------+---------------+
| ID | user_login       | display_name      | user_email                              | user_registered     | roles         |
+----+------------------+-------------------+-----------------------------------------+---------------------+---------------+
```

...then maybe this report output should also include the user_login, after the ID? 
(May be outside the scope of this PR)

**Review by GaryJones — COMMENTED** [review]
A couple of minor bits, but overall ✅ 

Would be good to add some Behat tests to check the guard clauses you have in here, as well successful output. ([example](https://github.com/Automattic/Co-Authors-Plus/blob/develop/features/rename-coauthor.feature))

Do you have any thoughts about getting this into https://github.com/WordPress/two-factor instead?

File: wp-cli/vip-two-factor.php (hunk 1 above)
Comment (rebeccahum) [reply wp-cli/vip-two-factor.php]: It includes secondary providers 🙂

File: wp-cli/vip-two-factor.php (hunk 3 above)
Comment (rebeccahum) [reply wp-cli/vip-two-factor.php:141]: Nah, I don't think it's necessary given the user lookup can be login/id/email.

**Pull request merged**